docs/release/README.md

```diff
@@ -0,0 +1,59 @@
+# Releasing the Training Operator
+
+## Prerequisite
+
+1. Permissions
+   - You need write permissions on the repository to create a release tag/branch.
+1. Prepare your Github Token
+
+1. Install Github python dependencies to generate changlog
+   ```
+   pip install PyGithub==1.55
+   ```
+
+### Release Process
+
+1. Make sure the last commit you want to release past `kubeflow-training-operator-postsubmit` testing.
+
+1. Check out that commit (in this example, we'll use `6214e560`).
+
+1. Depends on what version you want to release,
+
+   - Major or Minor version - Use the GitHub UI to cut a release branch and name the release branch `v{MAJOR}.${MINOR}-branch`
+   - Patch version - You don't need to cut release branch.
+
+1. Create a new PR against the release branch to change container image in manifest to point to that commit hash.
+
+   ```
+   images:
+   - name: kubeflow/training-operator
+     newName: kubeflow/training-operator
+     newTag: ${commit_hash}
+   ```
+
+   > note: post submit job will always build a new image using the `PULL_BASE_HASH` as image tag.
```

Comment:
```suggestion
   > [!NOTE]
   > Post submit job will always build a new image using the `PULL_BASE_HASH` as image tag.
```

Let's use the GitHub-style notification.
https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts